README.md

```diff
@@ -1,34 +1,249 @@
+token         |   yes    |                              | The token to be used for the service
+service       |    no    |                   'gh-pages' | The publishing service to deploy the site
+purpose       |    no    |                 'deployment' | The deployment purpose
+domain        |    no    |                           '' | The domain that the site is available at. Required if service chosen is surge
+version       |    no    |                     'latest' | The MarkBind version to use to build the site
+keepFiles     |    no    |                        false | Whether to keep the files in the published branch before pushing
+rootDirectory |    no    |                          '.' | The directory to read source files from
+baseUrl       |    no    | Value specified in site.json | The base URL relative to your domain
+siteConfig    |    no    |                  'site.json' | The site config file to use
+
+## Option Details
+
+### token
+Currently two types of tokens are supported (correspond to the two supported publishing services):
+- Token for GitHub Pages
+  - Simply use `${{ secrets.GITHUB_TOKEN }}`
+  - Note that you need to ensure that your have selected the branch that you want to deploy to in your GitHub repo's Pages settings
+- Token for Surge.sh
+  - Example: `${{ secrets.SURGE_TOKEN }}`
+    - `SURGE_TOKEN` is the environment secret name
+  - Require registration with an email address
+  - After retrieving the token, put the token as a repository secret in your
+  - See [here](https://markbind.org/userGuide/deployingTheSite.html#previewing-prs-using-surge) for a detailed guide on how to retrieve the token
+
+### service
+Currently two types of publishing services are supported:
+- GitHub Pages
+  - 'gh-pages'
+  - See [here](https://markbind.org/userGuide/deployingTheSite.html#publishing-to-github-pages) for more details
+- Surge.sh
+  - 'surge'
+  - See [here](https://surge.sh/) for more details
+
+### purpose
+The purpose of the deployment. This can be either standard deployment or for PR previewing.
+- Standard deployment
+  - 'deployment'
+  - This is the default value
+- PR preview
+  - 'pr-preview'
+  - This is used when you want to build and preview a site when there is a PR made to the repository. Note that this does not work for PR from a fork(due to security reasons)
+
+### domain
+The domain that the site is available at. Required if service chosen is surge. Surge.sh allows you to specify a subdomain as long as it is not taken up by others.
+
+- 'xxx.surge.sh'
+  - A typical domain that you can specify. You have to ensure that 'xxx' is unique. Read [here](https://surge.sh/help/adding-a-custom-domain) on how to configure a custom domain with Surge.sh
+- 'pr-x-domain'
+  - Note that for PR Preview purposes, the domain will be prefixed with 'pr-x', which 'x' is the GitHub event number
```

Comment:
One more question / suggestion: Is this action able to automatically cleanup merged PR deployments? If not, could also add a reminder on this to manually do so.

Reply:
For PR deployments (via Surge), the deployment will not be automatically deleted. This may be possible but the user will need to create another action to trigger the deletion on PR merge/close following this [instruction](https://surge.sh/help/tearing-down-a-project). I guess I can create that as a reusable workflow and link the usage here so that users can choose to include that additional step if needed.